taichi/runtime/llvm/runtime.cpp

```diff
@@ -1550,6 +1550,16 @@ void stack_push(Ptr stack, size_t max_num_elements, std::size_t element_size) {
 }
 
 #include "internal_functions.h"
+
+void set_partial_bits_b32(u32*ptr, u32 offset, u32 bits, u32 value) {
+  u32 mask = ~(u32)0;
+  mask = ((mask << offset) >> (32 - offset - bits));
```

Comment:
I'm a little confused here - if we are counting 0 from LSB, the correct mask should have `1`s starting from `offset` to `offset + bits`. That doesn't seem to be the case here.

One quick way to achieve the mask above seems to be `(((u32)1 << bits) - 1) << offset)`.